File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.py
def solution(numbers):
    answer = []
    for num in numbers:
        if num % 2 == 0:
            answer.append(num + 1)
        else:
            bin_num = list("0" + bin(num)[2:]) 
            idx = ''.join(bin_num).rfind("01") 
            bin_num[idx] = '1'
            bin_num[idx + 1] = '0'
            answer.append(int(''.join(bin_num), 2))
    return answer

File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.js
function solution(numbers) {
    let answer = [];
    for(const num of numbers){
        if(num % 2 === 0) answer.push(num + 1);
        else {
            let binNum = "0" + num.toString(2);
            const idx = binNum.lastIndexOf("01");
            answer.push(parseInt(binNum.slice(0, idx) + "10" + binNum.slice(idx + 2), 2));
        }
    }
    return answer;
}